src/scripts/main.js

```diff
@@ -1,5 +1,20 @@
 'use strict';
 
 document.addEventListener('click', (e) => {
-  // write code here
+  const wall = document.querySelector('.wall');
+  const spider = wall.querySelector('img');
+
+  const wallRect = wall.getBoundingClientRect();
+  const spiderWidth = spider.offsetWidth;
+  const spiderHeight = spider.offsetHeight;
+
+  let x = e.clientX - 5 - wallRect.left - spiderWidth / 2;
+  let y = e.clientY - wallRect.top - spiderHeight / 2;
+
+  x = Math.max(0, Math.min(x, wallRect.width - spiderWidth - 20));
+  y = Math.max(0, Math.min(y, wallRect.height - spiderHeight - 20));
```

Comment:
The boundary calculations subtract 20 from the wall's width and height. This seems to be a margin or padding value. Verify that this value is correct and consistent with your design requirements.